mne/stats/regression.py

```diff
@@ -310,6 +311,9 @@ def solver(X, Y):
 
         # this is the magical part (thanks to Marijn van Vliet):
         # use toeplitz to construct series of diagonals
+        # XXX this is magical but terribly inefficient as
+        # linalg.toeplitz(samples, lags) allocates a dense matrix with
+        # n_samples rows for each condition in the loop !!!
         X[:, cum:cum + tmax_ - tmin_] = linalg.toeplitz(samples, lags)
```

Comment:
I have code to construct X as a sparse matrix (faster & less memory). I'm struggling to solve the linear system properly. E.g., 

```
X
<211066x1270 sparse matrix of type '<class 'numpy.float64'>'
	with 1212715 stored elements in COOrdinate format>
data.shape
(64, 211066)
```

How do I get the coefficients?
MATLAB seemingly does QR factorization here (for sparse rectangular X), but from what I can tell, in scipy, this requires us to create a dense matrix from X. This works - I've tried - but it doesn't seem optimal.

As you know, Nathaniel Smith does incremental matrix inverses, see [here](https://github.com/rerpy/rerpy/blob/master/rerpy/rerp.py#L1329).